src/timeline/js/controllers.js

```diff
@@ -413,6 +438,25 @@ App.controller('TimelineCtrl',function($scope) {
 	 var new_cursor_x = Math.round((cursor_time * $scope.pixelsPerSecond) - center_x);
 	 $("#scrolling_tracks").scrollLeft(new_cursor_x);
  };
+ 
+ // Center the timeline on a given time position
+ $scope.centerOnTime = function(centerTime) {
+    // Get the width of the timeline
+    var scrollingTracksWidth = $("#scrolling_tracks").width();
+    
+    // Calculate the position to scroll the timeline to to center on the requested time
+    var pixelToCenterOn = parseFloat(centerTime) * $scope.pixelsPerSecond;
+    var scrollPosition = Math.max(pixelToCenterOn - (scrollingTracksWidth / 2.0), 0);
+    
+    // Scroll the timeline using JQuery
+    $("#scrolling_tracks").scrollLeft(Math.round(scrollPosition + 0.5));
+ };
```

Comment:
So, I do seem to have encountered one bug, and I _think_ this is the source.

If you have a clip on the timeline that starts right at frame 1 (time index `00:00:00:01`), and you use Prev Marker to jump to the left edge of the clip, the playhead line isn't actually visible, because it's scrolled just _one_ pixel off the left edge of the canvas.

I think that `Math.round(scrollPosition + 0.5)` is causing it to round up to 1, when we want the value to be 0.

Reply:
JS has a [`Math.floor()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/floor) as well — if the goal is just to offset the rounding for half-values, that should work. When you add `0.5`, any values >= `x.5` will get pushed over to `x+1`, but everything else will round down to `x`.

Reply:
Yes, that's what I *meant* to do.  Math.floor is definitely what I wanted.  Thank you.

Thank you for the patience and explanations.  It's not lost on me that it would have taken less time and effort for you to make these fixes yourself rather than pointing them out and explaining them to me.  Hopefully, this means my next attempt will be better.

Reply:
> It's not lost on me that it would have taken less time and effort for you to make these fixes yourself rather than pointing them out and explaining them to me.

Oh, not at all. Sure, maybe I _could_ write it, but the fact is that I didn't. You did! And we can always use (I'm trying not to say "desperately need", but... yeah, that) more people familiar with and working in the OpenShot code, so getting contributors up to speed is hardly an imposition.

Plus, honestly I can easily see myself making that same round-vs-floor mistake in my own code, and _not_ catching it in testing. It's always easier to take a step back and see things like that when it's someone else's code.

And that's kind of the whole idea behind code review — a new thing for me, but something I'm trying to embrace — really: that a second pair of eyes is helpful no matter who you are or how well you know the code. In fact, it only really works if all of the participants _could_ write the code.

(TBH, I probably wouldn't have thought to abstract out the visibility-detection logic from #2938 into a reusable `isTimeVisible()`, without this PR. And even if I had, I probably wouldn't have _bothered_. The code was **there** in my original PR, sure, but it's definitely better organized as a result of our discussions and brainstorming here.)